pkg/kubelet/kubelet.go

```diff
@@ -1580,31 +1584,27 @@ func (kl *Kubelet) SyncPods(allPods []api.Pod, podSyncTypes map[types.UID]metric
 
 	// Kill any containers we don't need.
 	killed := []string{}
-	for ix := range dockerContainers {
-		// Don't kill containers that are in the desired pods.
-		dockerName, _, err := dockertools.ParseDockerName(dockerContainers[ix].Names[0])
-		_, found := desiredPods[dockerName.PodUID]
-		if err == nil && found {
+	for _, pod := range runningPods {
```

Comment:
@dchen1107 I think the syncPod() will take care of the update case.

So the logic I want to express here is:

```
foreach desired pod:
    find its corresponding running pod, and do syncPod.
    
foreach running pod:
    if it is not desired, kill the whole pod.
```

The first loop makes sure we have all the pod up to the desired state. The second loop makes sure we remove any unnecessary pods.

The assumption here is all containers we created can be grouped into a pod.